diff --git a/node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial/BLE.h b/node_modules/react-native-bluetooth-serial-next/ios/BLE.h
similarity index 100%
rename from node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial/BLE.h
rename to node_modules/react-native-bluetooth-serial-next/ios/BLE.h
diff --git a/node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial/BLE.m b/node_modules/react-native-bluetooth-serial-next/ios/BLE.m
similarity index 100%
rename from node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial/BLE.m
rename to node_modules/react-native-bluetooth-serial-next/ios/BLE.m
diff --git a/node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial/BLEDefines.h b/node_modules/react-native-bluetooth-serial-next/ios/BLEDefines.h
similarity index 100%
rename from node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial/BLEDefines.h
rename to node_modules/react-native-bluetooth-serial-next/ios/BLEDefines.h
diff --git a/node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial/CBPeripheral+BTSExtensions.h b/node_modules/react-native-bluetooth-serial-next/ios/CBPeripheral+BTSExtensions.h
similarity index 100%
rename from node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial/CBPeripheral+BTSExtensions.h
rename to node_modules/react-native-bluetooth-serial-next/ios/CBPeripheral+BTSExtensions.h
diff --git a/node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial/CBPeripheral+BTSExtensions.m b/node_modules/react-native-bluetooth-serial-next/ios/CBPeripheral+BTSExtensions.m
similarity index 100%
rename from node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial/CBPeripheral+BTSExtensions.m
rename to node_modules/react-native-bluetooth-serial-next/ios/CBPeripheral+BTSExtensions.m
diff --git a/node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial/RCTBluetoothSerial.h b/node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial.h
similarity index 93%
rename from node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial/RCTBluetoothSerial.h
rename to node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial.h
index 5c53939..a11dfb0 100644
--- a/node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial/RCTBluetoothSerial.h
+++ b/node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial.h
@@ -11,8 +11,16 @@
  
  */
 
+#if __has_include(<React/RCTBridgeModule.h>)
 #import <React/RCTBridgeModule.h>
+#else
+#import "RCTBridgeModule.h"
+#endif
+#if __has_include(<React/RCTEventEmitter.h>)
 #import <React/RCTEventEmitter.h>
+#else
+#import "RCTEventEmitter.h"
+#endif
 #import "BLE.h"
 
 typedef void (^RCTBluetoothSerialPeripheralCallback) (CBPeripheral *_Nullable peripheral);
diff --git a/node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial/RCTBluetoothSerial.m b/node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial.m
similarity index 100%
rename from node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial/RCTBluetoothSerial.m
rename to node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial.m
diff --git a/node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial.xcodeproj/project.pbxproj b/node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial.xcodeproj/project.pbxproj
index 0c7b6b7..a2eb90b 100644
--- a/node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial.xcodeproj/project.pbxproj
+++ b/node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial.xcodeproj/project.pbxproj
@@ -7,10 +7,10 @@
 	objects = {
 
 /* Begin PBXBuildFile section */
-		69CA06071CC43C2900AF6479 /* RCTBluetoothSerial.h in Copy Files */ = {isa = PBXBuildFile; fileRef = 69CA06061CC43C2900AF6479 /* RCTBluetoothSerial.h */; };
-		69CA06091CC43C2900AF6479 /* RCTBluetoothSerial.m in Sources */ = {isa = PBXBuildFile; fileRef = 69CA06081CC43C2900AF6479 /* RCTBluetoothSerial.m */; };
-		69CA06161CC49D5E00AF6479 /* BLE.m in Sources */ = {isa = PBXBuildFile; fileRef = 69CA06101CC49D5E00AF6479 /* BLE.m */; };
-		69CA06171CC49D5E00AF6479 /* CBPeripheral+BTSExtensions.m in Sources */ = {isa = PBXBuildFile; fileRef = 69CA06131CC49D5E00AF6479 /* CBPeripheral+BTSExtensions.m */; };
+		04BE5B0523B25A0300B245DE /* BLE.m in Sources */ = {isa = PBXBuildFile; fileRef = 04BE5AFE23B25A0300B245DE /* BLE.m */; };
+		04BE5B0623B25A0300B245DE /* RCTBluetoothSerial.m in Sources */ = {isa = PBXBuildFile; fileRef = 04BE5AFF23B25A0300B245DE /* RCTBluetoothSerial.m */; };
+		04BE5B0723B25A0300B245DE /* CBPeripheral+BTSExtensions.m in Sources */ = {isa = PBXBuildFile; fileRef = 04BE5B0123B25A0300B245DE /* CBPeripheral+BTSExtensions.m */; };
+		04BE5B0823B25A1100B245DE /* RCTBluetoothSerial.h in Copy Files */ = {isa = PBXBuildFile; fileRef = 04BE5B0423B25A0300B245DE /* RCTBluetoothSerial.h */; };
 /* End PBXBuildFile section */
 
 /* Begin PBXCopyFilesBuildPhase section */
@@ -20,7 +20,7 @@
 			dstPath = "include/$(PRODUCT_NAME)";
 			dstSubfolderSpec = 16;
 			files = (
-				69CA06071CC43C2900AF6479 /* RCTBluetoothSerial.h in Copy Files */,
+				04BE5B0823B25A1100B245DE /* RCTBluetoothSerial.h in Copy Files */,
 			);
 			name = "Copy Files";
 			runOnlyForDeploymentPostprocessing = 0;
@@ -28,14 +28,14 @@
 /* End PBXCopyFilesBuildPhase section */
 
 /* Begin PBXFileReference section */
+		04BE5AFE23B25A0300B245DE /* BLE.m */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.objc; path = BLE.m; sourceTree = "<group>"; };
+		04BE5AFF23B25A0300B245DE /* RCTBluetoothSerial.m */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.objc; path = RCTBluetoothSerial.m; sourceTree = "<group>"; };
+		04BE5B0023B25A0300B245DE /* BLEDefines.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = BLEDefines.h; sourceTree = "<group>"; };
+		04BE5B0123B25A0300B245DE /* CBPeripheral+BTSExtensions.m */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.objc; path = "CBPeripheral+BTSExtensions.m"; sourceTree = "<group>"; };
+		04BE5B0223B25A0300B245DE /* BLE.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = BLE.h; sourceTree = "<group>"; };
+		04BE5B0323B25A0300B245DE /* CBPeripheral+BTSExtensions.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = "CBPeripheral+BTSExtensions.h"; sourceTree = "<group>"; };
+		04BE5B0423B25A0300B245DE /* RCTBluetoothSerial.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = RCTBluetoothSerial.h; sourceTree = "<group>"; };
 		69CA06031CC43C2900AF6479 /* libRCTBluetoothSerial.a */ = {isa = PBXFileReference; explicitFileType = archive.ar; includeInIndex = 0; path = libRCTBluetoothSerial.a; sourceTree = BUILT_PRODUCTS_DIR; };
-		69CA06061CC43C2900AF6479 /* RCTBluetoothSerial.h */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.h; path = RCTBluetoothSerial.h; sourceTree = "<group>"; };
-		69CA06081CC43C2900AF6479 /* RCTBluetoothSerial.m */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.objc; path = RCTBluetoothSerial.m; sourceTree = "<group>"; };
-		69CA060F1CC49D5E00AF6479 /* BLE.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = BLE.h; sourceTree = "<group>"; };
-		69CA06101CC49D5E00AF6479 /* BLE.m */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.objc; path = BLE.m; sourceTree = "<group>"; };
-		69CA06111CC49D5E00AF6479 /* BLEDefines.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = BLEDefines.h; sourceTree = "<group>"; };
-		69CA06121CC49D5E00AF6479 /* CBPeripheral+BTSExtensions.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = "CBPeripheral+BTSExtensions.h"; sourceTree = "<group>"; };
-		69CA06131CC49D5E00AF6479 /* CBPeripheral+BTSExtensions.m */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.objc; path = "CBPeripheral+BTSExtensions.m"; sourceTree = "<group>"; };
 /* End PBXFileReference section */
 
 /* Begin PBXFrameworksBuildPhase section */
@@ -52,7 +52,13 @@
 		69CA05FA1CC43C2900AF6479 = {
 			isa = PBXGroup;
 			children = (
-				69CA06051CC43C2900AF6479 /* RCTBluetoothSerial */,
+				04BE5B0223B25A0300B245DE /* BLE.h */,
+				04BE5AFE23B25A0300B245DE /* BLE.m */,
+				04BE5B0023B25A0300B245DE /* BLEDefines.h */,
+				04BE5B0323B25A0300B245DE /* CBPeripheral+BTSExtensions.h */,
+				04BE5B0123B25A0300B245DE /* CBPeripheral+BTSExtensions.m */,
+				04BE5B0423B25A0300B245DE /* RCTBluetoothSerial.h */,
+				04BE5AFF23B25A0300B245DE /* RCTBluetoothSerial.m */,
 				69CA06041CC43C2900AF6479 /* Products */,
 			);
 			sourceTree = "<group>";
@@ -65,20 +71,6 @@
 			name = Products;
 			sourceTree = "<group>";
 		};
-		69CA06051CC43C2900AF6479 /* RCTBluetoothSerial */ = {
-			isa = PBXGroup;
-			children = (
-				69CA060F1CC49D5E00AF6479 /* BLE.h */,
-				69CA06101CC49D5E00AF6479 /* BLE.m */,
-				69CA06111CC49D5E00AF6479 /* BLEDefines.h */,
-				69CA06121CC49D5E00AF6479 /* CBPeripheral+BTSExtensions.h */,
-				69CA06131CC49D5E00AF6479 /* CBPeripheral+BTSExtensions.m */,
-				69CA06061CC43C2900AF6479 /* RCTBluetoothSerial.h */,
-				69CA06081CC43C2900AF6479 /* RCTBluetoothSerial.m */,
-			);
-			path = RCTBluetoothSerial;
-			sourceTree = "<group>";
-		};
 /* End PBXGroup section */
 
 /* Begin PBXNativeTarget section */
@@ -118,6 +110,7 @@
 			developmentRegion = English;
 			hasScannedForEncodings = 0;
 			knownRegions = (
+				English,
 				en,
 			);
 			mainGroup = 69CA05FA1CC43C2900AF6479;
@@ -135,9 +128,9 @@
 			isa = PBXSourcesBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
-				69CA06091CC43C2900AF6479 /* RCTBluetoothSerial.m in Sources */,
-				69CA06161CC49D5E00AF6479 /* BLE.m in Sources */,
-				69CA06171CC49D5E00AF6479 /* CBPeripheral+BTSExtensions.m in Sources */,
+				04BE5B0623B25A0300B245DE /* RCTBluetoothSerial.m in Sources */,
+				04BE5B0523B25A0300B245DE /* BLE.m in Sources */,
+				04BE5B0723B25A0300B245DE /* CBPeripheral+BTSExtensions.m in Sources */,
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 		};
@@ -147,7 +140,7 @@
 		69CA060A1CC43C2900AF6479 /* Debug */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
-				ALWAYS_SEARCH_USER_PATHS = YES;
+				ALWAYS_SEARCH_USER_PATHS = NO;
 				CLANG_ANALYZER_NONNULL = YES;
 				CLANG_CXX_LANGUAGE_STANDARD = "gnu++0x";
 				CLANG_CXX_LIBRARY = "libc++";
@@ -201,7 +194,7 @@
 		69CA060B1CC43C2900AF6479 /* Release */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
-				ALWAYS_SEARCH_USER_PATHS = YES;
+				ALWAYS_SEARCH_USER_PATHS = NO;
 				CLANG_ANALYZER_NONNULL = YES;
 				CLANG_CXX_LANGUAGE_STANDARD = "gnu++0x";
 				CLANG_CXX_LIBRARY = "libc++";
@@ -249,7 +242,11 @@
 		69CA060D1CC43C2900AF6479 /* Debug */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
-				HEADER_SEARCH_PATHS = "$(SRCROOT)/../../react-native/React/**";
+				ALWAYS_SEARCH_USER_PATHS = NO;
+				HEADER_SEARCH_PATHS = (
+					"$(SRCROOT)/../../../React/**/**",
+					"$(SRCROOT)/../../react-native/React/**",
+				);
 				OTHER_LDFLAGS = "-ObjC";
 				PRODUCT_NAME = "$(TARGET_NAME)";
 				SKIP_INSTALL = YES;
@@ -259,7 +256,11 @@
 		69CA060E1CC43C2900AF6479 /* Release */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
-				HEADER_SEARCH_PATHS = "$(SRCROOT)/../../react-native/React/**";
+				ALWAYS_SEARCH_USER_PATHS = NO;
+				HEADER_SEARCH_PATHS = (
+					"$(SRCROOT)/../../../React/**/**",
+					"$(SRCROOT)/../../react-native/React/**",
+				);
 				OTHER_LDFLAGS = "-ObjC";
 				PRODUCT_NAME = "$(TARGET_NAME)";
 				SKIP_INSTALL = YES;
diff --git a/node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial.xcodeproj/project.xcworkspace/contents.xcworkspacedata b/node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial.xcodeproj/project.xcworkspace/contents.xcworkspacedata
deleted file mode 100644
index 919434a..0000000
--- a/node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial.xcodeproj/project.xcworkspace/contents.xcworkspacedata
+++ /dev/null
@@ -1,7 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<Workspace
-   version = "1.0">
-   <FileRef
-      location = "self:">
-   </FileRef>
-</Workspace>
diff --git a/node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial.xcodeproj/xcshareddata/xcschemes/RCTBluetoothSerial.xcscheme b/node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial.xcodeproj/xcshareddata/xcschemes/RCTBluetoothSerial.xcscheme
deleted file mode 100644
index 75752fb..0000000
--- a/node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial.xcodeproj/xcshareddata/xcschemes/RCTBluetoothSerial.xcscheme
+++ /dev/null
@@ -1,80 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<Scheme
-   LastUpgradeVersion = "1010"
-   version = "1.3">
-   <BuildAction
-      parallelizeBuildables = "NO"
-      buildImplicitDependencies = "YES">
-      <BuildActionEntries>
-         <BuildActionEntry
-            buildForTesting = "YES"
-            buildForRunning = "YES"
-            buildForProfiling = "YES"
-            buildForArchiving = "YES"
-            buildForAnalyzing = "YES">
-            <BuildableReference
-               BuildableIdentifier = "primary"
-               BlueprintIdentifier = "69CA06021CC43C2900AF6479"
-               BuildableName = "libRCTBluetoothSerial.a"
-               BlueprintName = "RCTBluetoothSerial"
-               ReferencedContainer = "container:RCTBluetoothSerial.xcodeproj">
-            </BuildableReference>
-         </BuildActionEntry>
-      </BuildActionEntries>
-   </BuildAction>
-   <TestAction
-      buildConfiguration = "Debug"
-      selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
-      selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
-      shouldUseLaunchSchemeArgsEnv = "YES">
-      <Testables>
-      </Testables>
-      <AdditionalOptions>
-      </AdditionalOptions>
-   </TestAction>
-   <LaunchAction
-      buildConfiguration = "Debug"
-      selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
-      selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
-      launchStyle = "0"
-      useCustomWorkingDirectory = "NO"
-      ignoresPersistentStateOnLaunch = "NO"
-      debugDocumentVersioning = "YES"
-      debugServiceExtension = "internal"
-      allowLocationSimulation = "YES">
-      <MacroExpansion>
-         <BuildableReference
-            BuildableIdentifier = "primary"
-            BlueprintIdentifier = "69CA06021CC43C2900AF6479"
-            BuildableName = "libRCTBluetoothSerial.a"
-            BlueprintName = "RCTBluetoothSerial"
-            ReferencedContainer = "container:RCTBluetoothSerial.xcodeproj">
-         </BuildableReference>
-      </MacroExpansion>
-      <AdditionalOptions>
-      </AdditionalOptions>
-   </LaunchAction>
-   <ProfileAction
-      buildConfiguration = "Release"
-      shouldUseLaunchSchemeArgsEnv = "YES"
-      savedToolIdentifier = ""
-      useCustomWorkingDirectory = "NO"
-      debugDocumentVersioning = "YES">
-      <MacroExpansion>
-         <BuildableReference
-            BuildableIdentifier = "primary"
-            BlueprintIdentifier = "69CA06021CC43C2900AF6479"
-            BuildableName = "libRCTBluetoothSerial.a"
-            BlueprintName = "RCTBluetoothSerial"
-            ReferencedContainer = "container:RCTBluetoothSerial.xcodeproj">
-         </BuildableReference>
-      </MacroExpansion>
-   </ProfileAction>
-   <AnalyzeAction
-      buildConfiguration = "Debug">
-   </AnalyzeAction>
-   <ArchiveAction
-      buildConfiguration = "Release"
-      revealArchiveInOrganizer = "YES">
-   </ArchiveAction>
-</Scheme>
diff --git a/node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial.xcworkspace/contents.xcworkspacedata b/node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial.xcworkspace/contents.xcworkspacedata
new file mode 100644
index 0000000..91379a8
--- /dev/null
+++ b/node_modules/react-native-bluetooth-serial-next/ios/RCTBluetoothSerial.xcworkspace/contents.xcworkspacedata
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Workspace version="1.0">
+  <FileRef location="group:BluetoothSerialNext.xcodeproj"></FileRef>
+</Workspace>
\ No newline at end of file
diff --git a/node_modules/react-native-bluetooth-serial-next/react-native-bluetooth-serial-next.podspec b/node_modules/react-native-bluetooth-serial-next/react-native-bluetooth-serial-next.podspec
new file mode 100644
index 0000000..7cba3cb
--- /dev/null
+++ b/node_modules/react-native-bluetooth-serial-next/react-native-bluetooth-serial-next.podspec
@@ -0,0 +1,21 @@
+require "json"
+
+package = JSON.parse(File.read(File.join(__dir__, "package.json")))
+
+Pod::Spec.new do |s|
+  s.name         = "react-native-bluetooth-serial-next"
+  s.version      = package["version"]
+  s.summary      = package["description"]
+  s.description  = <<-DESC
+                  react-native-bluetooth-serial-next
+                   DESC
+  s.homepage     = package["homepage"]
+  s.license      = { :type => "MIT", :file => "LICENSE" }
+  s.authors      = package["author"]
+  s.platforms    = { :ios => "9.0" }
+  s.source       = { :git => "https://github.com/nuttawutmalee/react-native-bluetooth-serial-next.git", :tag => "#{s.version}" }
+
+  s.dependency "React"
+
+  s.source_files = "ios/**/*.{h,m,swift}"
+end
\ No newline at end of file
